File: sdk/clojure/src/hitechcloud/sdk/affiliates_advanced.clj
(ns hitechcloud.sdk.affiliates_advanced
  (:require [hitechcloud.sdk.http-client :as http]))

;; AffiliatesAdvanced: advanced affiliate management with client_id

(defn get_info "GET /api/affiliates_adv/{client_id}/info - Get affiliate account info" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/info" params))

(defn get_stats "GET /api/affiliates_adv/{client_id}/stats - Get affiliate stats" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/stats" params))

(defn get_commission_plans "GET /api/affiliates_adv/{client_id}/commission-plans - Get commission plans" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/commission-plans" params))

(defn get_vouchers "GET /api/affiliates_adv/{client_id}/vouchers - Get vouchers" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/vouchers" params))

(defn get_commissions "GET /api/affiliates_adv/{client_id}/commissions - Get commissions" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/commissions" params))

(defn get_referrals "GET /api/affiliates_adv/{client_id}/referrals - Get referred clients" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/referrals" params))

(defn get_payouts "GET /api/affiliates_adv/{client_id}/payouts - Get payout history" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/payouts" params))

(defn get_campaigns "GET /api/affiliates_adv/{client_id}/campaigns - Get campaigns" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/campaigns" params))

(defn get_audit_log "GET /api/affiliates_adv/{client_id}/audit - Get activity log" [client & [params]] (http-get client "/api/affiliates_adv/{client_id}/audit" params))

(defn activate "POST /api/affiliates_adv/{client_id}/activate - Activate affiliate account" [client & [params]] (http-post client "/api/affiliates_adv/{client_id}/activate" params))

(defn set_commission_plan "POST /api/affiliates_adv/{client_id}/commission-plan/{commission_id} - Set commission plan" [client & [params]] (http-post client "/api/affiliates_adv/{client_id}/commission-plan/{commission_id}" params))

(defn create_voucher "POST /api/affiliates_adv/{client_id}/vouchers/{plan_id} - Create voucher" [client & [params]] (http-post client "/api/affiliates_adv/{client_id}/vouchers/{plan_id}" params))

(defn delete_voucher "DELETE /api/affiliates_adv/{client_id}/vouchers/{voucher_id} - Delete voucher" [client & [params]] (http-delete client "/api/affiliates_adv/{client_id}/vouchers/{voucher_id}" params))

(defn set_landing_page "POST /api/affiliates_adv/{client_id}/landing-page - Set landing page" [client & [params]] (http-post client "/api/affiliates_adv/{client_id}/landing-page" params))

File: sdk/fsharp/src/HiTechCloudSDK/AffiliatesAdvanced.fs
namespace HiTechCloud.SDK

/// AffiliatesAdvanced: advanced affiliate management with client_id
type AffiliatesAdvancedResource(httpClient: HttpClient) =

    /// GET /api/affiliates_adv/{client_id}/info - Get affiliate account info
    member this.GetInfo() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/info")
        }

    /// GET /api/affiliates_adv/{client_id}/stats - Get affiliate stats
    member this.GetStats() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/stats")
        }

    /// GET /api/affiliates_adv/{client_id}/commission-plans - Get commission plans
    member this.GetCommissionPlans() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/commission-plans")
        }

    /// GET /api/affiliates_adv/{client_id}/vouchers - Get vouchers
    member this.GetVouchers() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/vouchers")
        }

    /// GET /api/affiliates_adv/{client_id}/commissions - Get commissions
    member this.GetCommissions() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/commissions")
        }

    /// GET /api/affiliates_adv/{client_id}/referrals - Get referred clients
    member this.GetReferrals() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/referrals")
        }

    /// GET /api/affiliates_adv/{client_id}/payouts - Get payout history
    member this.GetPayouts() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/payouts")
        }

    /// GET /api/affiliates_adv/{client_id}/campaigns - Get campaigns
    member this.GetCampaigns() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/campaigns")
        }

    /// GET /api/affiliates_adv/{client_id}/audit - Get activity log
    member this.GetAuditLog() =
        async {
            return! httpClient.getAsync("/api/affiliates_adv/{client_id}/audit")
        }

    /// POST /api/affiliates_adv/{client_id}/activate - Activate affiliate account
    member this.Activate() =
        async {
            return! httpClient.postAsync("/api/affiliates_adv/{client_id}/activate")
        }

    /// POST /api/affiliates_adv/{client_id}/commission-plan/{commission_id} - Set commission plan
    member this.SetCommissionPlan() =
        async {
            return! httpClient.postAsync("/api/affiliates_adv/{client_id}/commission-plan/{commission_id}")
        }

    /// POST /api/affiliates_adv/{client_id}/vouchers/{plan_id} - Create voucher
    member this.CreateVoucher() =
        async {
            return! httpClient.postAsync("/api/affiliates_adv/{client_id}/vouchers/{plan_id}")
        }

    /// DELETE /api/affiliates_adv/{client_id}/vouchers/{voucher_id} - Delete voucher
    member this.DeleteVoucher() =
        async {
            return! httpClient.deleteAsync("/api/affiliates_adv/{client_id}/vouchers/{voucher_id}")
        }

    /// POST /api/affiliates_adv/{client_id}/landing-page - Set landing page
    member this.SetLandingPage() =
        async {
            return! httpClient.postAsync("/api/affiliates_adv/{client_id}/landing-page")
        }

